test/e2e_federation/authn.go

```diff
@@ -35,16 +35,20 @@ var _ = framework.KubeDescribe("[Feature:Federation]", func() {
 	f := fedframework.NewDefaultFederatedFramework("federation-apiserver-authn")
 
 	var _ = Describe("Federation API server authentication [NoCluster]", func() {
+		var (
+			clusters fedframework.ClusterSlice
+		)
 		BeforeEach(func() {
 			fedframework.SkipUnlessFederated(f.ClientSet)
+			clusters = fedframework.ClusterSlice{}
```

Comment:
here you are initializing the clusters with 0 clusters which is not the correct thing to do. In this case, the validation to see if the service was created in federated cluster would not work as it won't enter the for loop of clusters to check at all.
Instead of passing clusters from every caller. please call
```
clusters = f.GetRegisteredClusters()
```
in createService itself.